src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -611,6 +611,15 @@ class ArrayLiteralSegment(BaseSegment):
     )
 
 
+class TimeZoneGrammar(BaseSegment):
+    """Casting to Time Zone."""
+
+    type = "time_zone_grammar"
+    match_grammar = AnyNumberOf(
+        Sequence("AT", "TIME", "ZONE", Ref("ExpressionSegment")),
```

Comment:
Seems like ColumnReferenceSegment and QuotedLiteralSegment are the only options here? Can we stick in a OneOf() for them, and kill the Expression?

Reply:
Ah fair 👍